packages/@react-aria/focus/test/FocusScopeOwnerDocument.test.js

```diff
@@ -0,0 +1,356 @@
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {act, fireEvent, pointerMap, render, waitFor} from '@react-spectrum/test-utils';
+import {FocusScope, useFocusManager} from '../';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import userEvent from '@testing-library/user-event';
+
+describe('FocusScope', function () {
+  let iframe;
+  let iframeRoot;
+
+  let user;
+
+  const IframeExample = ({children}) => {
+    React.useEffect(() => {
+      ReactDOM.render(<>{children}</>, iframeRoot);
+    }, [children]);
+
+    return null;
+  };
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+
+    // Iframe setup
+    iframe = document.createElement('iframe');
+    window.document.body.appendChild(iframe);
+    const iframeDocument = iframe.contentWindow.document;
+    iframeRoot = iframeDocument.createElement('div');
+    iframeDocument.body.appendChild(iframeRoot);
+  });
+
+  afterEach(() => {
+    // make sure to clean up any raf's that may be running to restore focus on unmount
+    act(() => {jest.runAllTimers();});
+
+    // Iframe teardown
+    iframe.remove();
+  });
+
+  describe('focus containment', function () {
+    it('should contain focus within the scope', async function () {
+      render(
+        <IframeExample>
+          <FocusScope contain>
+            <input data-testid="input1" />
+            <input data-testid="input2" />
+            <input data-testid="input3" />
+          </FocusScope>
+        </IframeExample>
+      );
+
+      await waitFor(() => {
+        expect(document.querySelector('iframe').contentWindow.document.body.querySelector('input[data-testid="input1"]')).toBeTruthy();
+      });
+
+      const iframeDocument = iframe.contentWindow.document;
+      user = userEvent.setup({delay: null, pointerMap, document: iframeDocument});
+      const input1 = iframeDocument.querySelector('input[data-testid="input1"]');
+      const input2 = iframeDocument.querySelector('input[data-testid="input2"]');
+      const input3 = iframeDocument.querySelector('input[data-testid="input3"]');
+
+      act(() => {input1.focus();});
+      expect(iframeDocument.activeElement).toBe(input1);
+
+      await user.tab();
+      expect(iframeDocument.activeElement).toBe(input2);
+
+      await user.tab();
+      expect(iframeDocument.activeElement).toBe(input3);
+
+      await user.tab();
+      expect(iframeDocument.activeElement).toBe(input1);
+
+      await user.tab({shift: true});
+      expect(iframeDocument.activeElement).toBe(input3);
+
+      await user.tab({shift: true});
+      expect(iframeDocument.activeElement).toBe(input2);
+
+      await user.tab({shift: true});
+      expect(iframeDocument.activeElement).toBe(input1);
+    });
+
+    it('uses document.activeElement instead of e.relatedTarget on blur to determine if focus is still in scope', async function () {
```

Comment:
what change is this referencing? and what is it actually testing? the test name shouldn't be about the implementation detail but instead about the feature it's supporting if possible

Reply:
It's this logic here! https://github.com/adobe/react-spectrum/commit/5919197d9e7bd9596cdb8e77e017de2a8415d731. It's to ensure that focusing within an iframe is retained. I'll change the test case name here.

Reply:
Ahhh, so sorry, didn't make the connection to the inspiration for this test. We named that one poorly. Would you mind updating the original to "focus properly moves into child iframe on click" instead? We can leave a comment on it as well "the test setup is a bit contrived to just purely simulate the blur/focus events that would happen in a case like this"

Note, the iframe in that case was a use case we came across where someone had put an iframe to some site (Wikipedia for example) in a dialog.

So this iframe case is probably a little different than yours, but you may want to consider if that has any bearing or implications on your work. ie, is this support in conflict? I don't see anything, however, we're still only just thinking about these iframes of yours and don't know all the inner workings/assumptions.

I think your test is still valid, though the name should be updated to match.